include/crocoddyl/multibody/actions/constraint-fwddyn.hpp

```diff
@@ -0,0 +1,150 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, INRIA
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MULTIBODY_ACTIONS_CONTACT_FWDDYN2_HPP_
+#define CROCODDYL_MULTIBODY_ACTIONS_CONTACT_FWDDYN2_HPP_
+
+#if PINOCCHIO_VERSION_AT_LEAST(2, 9, 0)
+
+#include <stdexcept>
+
+#include <pinocchio/algorithm/contact-info.hpp>
+#include <pinocchio/algorithm/proximal.hpp>
+#include <pinocchio/algorithm/constrained-dynamics.hpp>
+#include <pinocchio/algorithm/constrained-dynamics-derivatives.hpp>
+#include <pinocchio/algorithm/compute-all-terms.hpp>
+#include <pinocchio/algorithm/frames.hpp>
+#include <pinocchio/algorithm/centroidal.hpp>
+#include <pinocchio/algorithm/rnea.hpp>
+#include <pinocchio/algorithm/rnea-derivatives.hpp>
+#include <pinocchio/algorithm/kinematics-derivatives.hpp>
+
+#include "crocoddyl/core/utils/exception.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+#include "crocoddyl/multibody/fwd.hpp"
+#include "crocoddyl/multibody/data/constraint.hpp"
+#include "crocoddyl/core/diff-action-base.hpp"
+#include "crocoddyl/multibody/states/multibody.hpp"
+#include "crocoddyl/multibody/actuations/floating-base.hpp"
+#include "crocoddyl/core/costs/cost-sum.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class DifferentialActionModelConstraintFwdDynamicsTpl : public DifferentialActionModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef DifferentialActionModelAbstractTpl<Scalar> Base;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef CostModelSumTpl<Scalar> CostModelSum;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
+  typedef ActuationModelAbstractTpl<Scalar> ActuationModelAbstract;
+  typedef DifferentialActionDataAbstractTpl<Scalar> DifferentialActionDataAbstract;
+  typedef DifferentialActionDataConstraintFwdDynamicsTpl<Scalar> DifferentialActionDataConstraintFwdDynamics;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+  typedef pinocchio::RigidConstraintModelTpl<Scalar, 0> RigidConstraintModel;
+  typedef pinocchio::ProximalSettingsTpl<Scalar> ProximalSettings;
+
+  DifferentialActionModelConstraintFwdDynamicsTpl(
+      boost::shared_ptr<StateMultibody> state, boost::shared_ptr<ActuationModelAbstract> actuation,
+      const PINOCCHIO_STD_VECTOR_WITH_EIGEN_ALLOCATOR(RigidConstraintModel) & contacts,
```

Comment:
It is not clear to me how we could disable/enable constraints without going through the creating of a different model and its data. This feature is really important for MPC applications.